core-bundle/contao/pages/PageRegular.php

```diff
@@ -621,7 +619,20 @@ protected function createFooterScripts($objPage, $objLayout)
 		// Add the custom JavaScript
 		if ($objLayout->script)
 		{
-			$strScripts .= "\n" . trim($objLayout->script) . "\n";
+			$customScript = trim($objLayout->script);
+
+			if ($this->responseContext->has(CSPBuilder::class))
+			{
+				/** @var CSPBuilder $csp */
+				$csp = $this->responseContext->get(CSPBuilder::class);
+				
+				if ($nonce = $csp->nonce('script-src'))
+				{
+					$customScript = str_replace('<script', '<script nonce="' . $nonce . '"', $customScript);
+				}
+			}
+
+			$strScripts .= "\n" . $customScript . "\n";
```

Comment:
Arguably we could omit this entirely and just  have a known limitation that you cannot use custom JavaScripts in your page layout if you use any `script-src` (or `default-src`) policy (as your script will not have a valid nonce then). @contao/developers wdyt?

Reply:
I’m not sure. On the one hand, I think using string replacement to add the nonce should be avoided as much as possible, but on the other hand `$objLayout->script` is considered to be trusted so it would be nice to keep this code.